### PR TITLE
add known limitations and code decisions

### PR DESCRIPTION
**Pull request by r-f-g**
This PR provides "Known limitations" and "Code design decisions".

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -27,15 +27,34 @@ run the following.
...
-```
+   Alternatively, a machine can be targeted:
+
+   ```bash
+   $ juju-verify shutdown --machines 0
+   ```
+
+2. If you run a check on unit, which contains a subordinate unit, you will got only
+   a warning message about existence of the subordinate unit. To check if it is safe
+   to remove this unit, you must perform a check on this unit or if the check is not
```
Comment (auria) [inline README.md:44]: "In order to check if it is safe to remove this unit, juju-verify needs to be explictly run against this subordinate unit, or the unit needs to be manually checked (if juju-verify does not support this charm yet)"

File: README.md [hunk 2]
```diff
@@ -27,15 +27,34 @@ run the following.
 $ juju-verify shutdown --units nova-compute/0 nova-compute/1
 ```
 
-**NOTE:** If you run check on multiple units at the same time, they must all run
-the same charm. Trying, for example, `juju-verify shutdown --units nova-compute/1
-ceph-osd/0` will result in error.
+## Known limitations
 
-Alternatively, a machine can be targeted:
+1. If you run check on multiple units at the same time, they must all run
```
Comment (auria) [inline README.md:32]: "If a check is run on multiple units at the same time, ..."

File: README.md [hunk 3]
```diff
@@ -58,8 +77,90 @@ make test
...
+* `NAME` - name of verifier
+* `verify_<action-with-unit>` - function to run all necessary checks when trying to 
+	                          perform "action" with the unit
+
+Each verifier will contain these two variables:
+
+* units - list of units we want to verify
+* model - corresponding model containing units
+
+The verifier should run verification using the `verify` function, which will check
```
Comment (auria) [inline README.md:104]: "should run verifications" (or "a verification").

**Review by auria — CHANGES_REQUESTED** [review]
Some suggestions around wording. Structure and content look good, thanks!

**Review by auria — APPROVED** [review]
LGTM. I've added a suggestion for a minor fix, but it's not a blocker.

File: README.md [hunk 4]
```diff
@@ -37,14 +37,132 @@ run the following.
...
+environment variable is set to a suitable VIP address before running functional tests.
+
+## Code decisions
+
+The main idea of `juju-verify` is to be used as a CLI tool with an entry point defined
+in `juju verify/juju verify.py`. We use the `argparse` library to parse CLI arguments
+and to provide help information, which can be accessed using `juju-verify --help`
+command.
+
+Despite the main purpose, it is possible to use `juju-verify` as python pacakge. It
```
[the remark this answers is not in the document]
Comment (r-f-g) [reply README.md]: Fixed while I merge the master branch.

**Review by r-f-g — COMMENTED** [review]
recheck-snap

**Review by auria — APPROVED** [review]
+1, ty

**Pull request merged**